app/src/test/java/com/example/bottomnav/ParseTest.java

```diff
@@ -109,82 +103,130 @@ public void structNConst() throws Exception {
 
         assertEquals("voltage_be", struct.get(0).name);
         assertEquals("current_be", struct.get(1).name);
         assertEquals("control", struct.get(2).name);
         assertEquals("reserved1", struct.get(3).name);
         assertEquals("reserved2", struct.get(4).name);
         assertEquals("reserved3", struct.get(5).name);
-        assertEquals("uint16_t", struct.get(0).type);
-        assertEquals("uint16_t", struct.get(1).type);
-        assertEquals("uint8_t", struct.get(2).type);
-        assertEquals("uint8_t", struct.get(3).type);
-        assertEquals("uint8_t", struct.get(4).type);
-        assertEquals("uint8_t", struct.get(5).type);
+        assertEquals("uint16_t", struct.get(0).payloadDataType);
+        assertEquals("uint16_t", struct.get(1).payloadDataType);
+        assertEquals("uint8_t", struct.get(2).payloadDataType);
+        assertEquals("uint8_t", struct.get(3).payloadDataType);
+        assertEquals("uint8_t", struct.get(4).payloadDataType);
+        assertEquals("uint8_t", struct.get(5).payloadDataType);
     }
 
     @Test
     public void checkIDToStructMap() throws Exception {
         Parse test = Parse.parseTextFile("parseData.h");
-
-        assertEquals(test.getStruct("ChargerStatusStruct"),
+        
+        assertEquals(test.getStructContents("ChargerStatusStruct"),
                 test.getCanStruct("CAN_CHARGER_STATUS"));
-        assertEquals(test.getStruct("ChargerControlStruct"),
+        assertEquals(test.getStructContents("ChargerControlStruct"),
                 test.getCanStruct("CAN_CHARGER_CONTROL"));
-        assertEquals(test.getStruct("CanStrainGaugeStruct"),
+        assertEquals(test.getStructContents("CanStrainGaugeStruct"),
                 test.getCanStruct("CAN_STRAIN_DATA"));
-        assertEquals(test.getStruct("CanPedalPosStruct"),
+        assertEquals(test.getStructContents("CanPedalPosStruct"),
                 test.getCanStruct("CAN_PEDAL_POS"));
     }
+
+    @Test
+    public void decodingSimple() throws Exception {
+        Parse test = Parse.parseTextFile("decode.h");
+
+        byte[] packedFloatPayload = {0x71, (byte) 0xFD, 0x47, 0x41};
+        String packedFloatMessage = "single PACKED_FLOAT, float: 12.499375";
+        System.out.println(test.decode(0x310, packedFloatPayload));
+        assertEquals(packedFloatMessage, test.decode(0x310, packedFloatPayload));
+        IntegerDecoder floatDec = (IntegerDecoder) test.getDecoder(0x310);
+        assertEquals(12.499375343322754,(float) floatDec.valueToRaw(), 0.000000000000001);
+
+        byte[] var4Payload = {0x0B, 0x00, 0x00, (byte) 0xA5};
+        String var4PaylodMessage = "struct CanPedalPosStruct, \n" +
+                "    accelPos, uint8_t: 11\n" +
+                "    brakePos, uint8_t: 0\n" +
+                "    reserved1Pos, uint8_t: 0\n" +
+                "    reserved2Pos, uint8_t: 165";
+        System.out.println(test.decode(0x282, var4Payload));
```

Comment:
You generally don't want printlns in your test code. It's fine for debugging, but unit tests are meant to be run fully automated aside from the pass/fail at the end. tl;dr remove stray debugging printlns in final code